### PR TITLE
Add Navtrail with dropdown

### PR DESCRIPTION
**Pull request by omar2205**
This adds a dropdown menu and a user avatar

Not logged
image

Logged in
image

### CODE REVIEW

**Comment by omar2205** [comment]
big oof 😛

**Comment by omar2205** [comment]
@w3cj this error
```
Error:   10:23  error  Parse errors in imported module './NavTrail.svelte': '>' expected. (1:8)
import/no-named-as-default
```
is related to #37 issue

**Comment by w3cj** [comment]
I think you will need to merge with main for the CI checks to pass.

**Comment by omar2205** [comment]
I just have to do another commit for it to recheck it

File: src/routes/+layout.svelte [hunk 1]
```diff
@@ -2,11 +2,13 @@
 	import '@skeletonlabs/skeleton/themes/theme-crimson.css';
 	import '@skeletonlabs/skeleton/styles/all.css';
 	import '../app.postcss';
-	import { AppBar, AppShell, LightSwitch } from '@skeletonlabs/skeleton';
-	import { signOut } from '@auth/sveltekit/client';
+	import { AppBar, AppShell } from '@skeletonlabs/skeleton';
 	import { setLocale } from '$lib/i18n/i18n-svelte.js';
 	import type { LayoutData } from './$types.js';
 
+	// zeslint-disable-next-line import/no-named-as-default-member
```
Comment (w3cj) [inline src/routes/+layout.svelte]: Should be able to remove this comment.

File: src/routes/NavTrail.svelte [hunk 2]
```diff
@@ -0,0 +1,90 @@
+<script lang="ts">
+	import { Avatar, LightSwitch, menu, storeLightSwitch } from '@skeletonlabs/skeleton';
+	import { signOut } from '@auth/sveltekit/client';
+
+	// Remove after PR#32
+	// eslint-disable-next-line import/no-unresolved
```
Comment (w3cj) [inline src/routes/NavTrail.svelte:6]: #32 should have fixed this... we might need to modify the $app alias

Comment (w3cj) [reply src/routes/NavTrail.svelte:6]: Remove theses comments for this PR

File: src/routes/NavTrail.svelte [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+	import { page } from '$app/stores';
+	// eslint-disable-next-line import/no-unresolved
```
Comment (w3cj) [inline src/routes/NavTrail.svelte:8]: Remove this comment.

File: src/routes/+layout.svelte [hunk 4]
```diff
@@ -2,11 +2,13 @@
...
+	import NavTrail from './NavTrail.svelte';
```
Comment (w3cj) [inline src/routes/+layout.svelte:10]: WHAT???

**Review by w3cj — CHANGES_REQUESTED** [review]

**Comment by brilvio** [comment]
Your branch is 3 commits behind main, maybe rebase or merge main into your branch.

File: src/routes/+layout.svelte (hunk 4 above)
Comment (omar2205) [reply src/routes/+layout.svelte:10]: I had an error if I didn't remove this, was testing if I "removed" it by adding z it shouldn't look at it.

**Comment by w3cj** [comment]
Regarding the $app multiple import error, I think the fix in the tsconfig.json might have been wrong. Can you try removing the $app aliases in there and add "./node_modules/@sveltejs/kit/types/ambient.d.ts" to the includes array? I'm not near a PC to try right now.

**Comment by omar2205** [comment]
removing the tsconfig fix results in `import/no-unresolved`

**Comment by w3cj** [comment]
Please merge / rebase with master. I updated the eslint config to handle multiple $app imports.

**Comment by omar2205** [comment]
Still, we get `import/no-named-as-default-member`

**Comment by w3cj** [comment]
Looks like the rule `import/no-named-as-default-member` is the issue instead of `import/import/no-named-as-default`??

**Comment by omar2205** [comment]
Not sure if it's both or just one of them.

**Comment by w3cj** [comment]
Let's disable these rules for .svelte files specifically. Do you mind opening another PR?

This section in the eslint config can be updated and we can remove `'import/no-named-as-default': 0,` in the general rules:

```js
	overrides: [
		{
			files: ['**/*.svelte'],
			parser: 'svelte-eslint-parser',
			parserOptions: {
				parser: '@typescript-eslint/parser',
			},
			rules: {
				'import/no-named-as-default': 0,
				'import/no-named-as-default-member': 0,
			},
		},
	],
```

**Comment by omar2205** [comment]
btw, I keep getting this warning
```
`parseForESLint` from parser `@typescript-eslint/parser` is invalid and will just be ignored
```

**Pull request merged**